### PR TITLE
fix(ecocredit): return grpc errors in basket and marketplace queries

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Ref: #1577 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -188,3 +188,5 @@ replace github.com/regen-network/regen-ledger/errors => ./errors
 replace github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.7.0
 
 replace github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
+
+replace github.com/regen-network/regen-ledger/errors => ./errors
```
Comment (aleem1314) [inline go.mod:192]:
```suggestion
```

File: x/ecocredit/basket/keeper/query_fees.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func (k Keeper) BasketFee(ctx context.Context, req *types.QueryBasketFeeRequest)
 		var ok bool
 		fee, ok = regentypes.ProtoCoinToCoin(basketFee.Fee)
 		if !ok {
-			return nil, sdkerrors.ErrInvalidType.Wrapf("failed to parse basket fee")
+			return nil, regenerrors.ErrInvalidArgument.Wrap("failed to parse basket fee")
```
Comment (ryanchristo) [inline x/ecocredit/basket/keeper/query_fees.go:25]: Should this be internal? The user is not providing an invalid argument.

```suggestion
			return nil, regenerrors.ErrInternal.Wrap("failed to parse basket fee")
```

**Review by ryanchristo — APPROVED** [review]
LGTM. One question/suggestion.

**Review by technicallyty — APPROVED** [review]
LGTM - preapproving, pending Ryan's suggestion

**Pull request merged**